docs/tutorials/typescript/workqueue-slackbot/code/temporal-application/src/workflows/workqueue.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  CancellationScope,
+  CancelledFailure,
+  continueAsNew,
+  defineQuery,
+  defineSignal,
+  setHandler,
+  workflowInfo,
+} from "@temporalio/workflow";
+import {WorkqueueData} from "../../../common-types/types";
+
+const getWorkqueueDataQuery = defineQuery<WorkqueueData[]>("getWorkqueueData");
+const addWorkqueueDataSignal =
+  defineSignal<[WorkqueueData]>("addWorkqueueData");
+const claimWorkSignal =
+  defineSignal<[{workId: string; claimantId: string}]>("claimWork");
+const completeWorkSignal = defineSignal<[{workId: string}]>("completeWork");
+
+export async function workqueue(existingData?: WorkqueueData[]): Promise<void> {
+  const wqdata: WorkqueueData[] = existingData ?? [];
+
+  while (!workflowInfo().continueAsNewSuggested) {
+    // Register a Query handler for 'getWorkqueueData'
+    setHandler(getWorkqueueDataQuery, () => {
```

Comment:
I don't think you want to register handlers in a loop.

Reply:
Ah, ya that makes sense.